browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -48,6 +49,16 @@
 
 namespace brave {
 
+namespace {
+
+const std::string& GetCanonicalName(
+    const std::vector<std::string>& dns_aliases) {
+  static const base::NoDestructor<std::string> nullstring_result;
+  return dns_aliases.size() >= 1 ? dns_aliases.front() : *nullstring_result;
```

Comment:
@mariospr I think we should be checking for `==1` here. In line 103 we set `optional_parameters->include_canonical_name = true;` and the https://source.chromium.org/chromium/chromium/src/+/main:net/dns/host_resolver.h;l=268 comment suggest only 1 result should be returned in this case (also see the comment here https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/pepper/pepper_host_resolver_message_filter.cc;l=192)

Reply:
That change seemed to be needed not to get the following tests failing:
```
AdBlockServiceTest.CnameCloakedRequestsCanBeExcepted     AdBlockServiceTest.CnameCloakedRequestsGetBlocked 
```
This said, and based on your comments, It's unclear to me yet why that's the case but debugging this I can see how this helper function gets called three times with each of those tests, having `dns_aliases` contain 1 element on the first 2 times, and 3 elements on the third time, which is why the `>=` seems to fix the tests for now.

I'll take a deeper look tomorrow to try to understand better whether there's a bug upstream or maybe something else we need to do in the tests to try to be good with `== 1`, maybe it's related to the MockHostResolver being used.

Reply:
I did some more investigation yesterday and it looks like this is happening because the tests don't use the internal hsot resolver but the test-only MockHostResolver, which doesn't really account for the `include_canonical_name` option yet (looks like it's WIP according to the comments in https://crrev.com/c/3352433 and the state (still open) of the associated bug crbug.com/1264933.

Talked to @iefremov and he thinks it's ok to leave this as-is for now.